cloud/pkg/cloudhub/channelq/channelq.go

```diff
@@ -100,9 +100,10 @@ func (q *ChannelMessageQueue) addMessageToQueue(nodeID string, msg *beehiveModel
 		return
 	}
 
-	item, exist, _ := nodeStore.GetByKey(messageKey)
-
-	if !isDeleteMessage(msg) {
+	//if the operation is delete, force to sync the resource message
+	//if the operation is response, force to sync the resource message, since the edgecore requests it
+	if !isDeleteMessage(msg) && msg.GetOperation() != beehiveModel.ResponseOperation {
```

Comment:
Thanks  for the fixing!

Here the `if !isDeleteMessage(msg)` means if the message is not delete message, we will do some check like remove the dumplicate message. And if the message is delete message, we will send it directly.

And the Response Message we will also send to edge at here: https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/cloudhub/channelq/channelq.go#L168

So this is not the root cause of the issue.

Reply:
Yes, if `exist` is true which means the duplicate message in nodeStore, as you mentioned,  some checks will be done like remove the dumplicate message. But if `exist` is false, I see `objectSync` may prevent the message to be send to edge: https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/cloudhub/channelq/channelq.go#L116.

And as a reponse to query a secret,  the field `content` stores the replied secret object, not string `OK`, so the boolean value of the experession at https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/cloudhub/channelq/channelq.go#L170 will be false which means the message is not list resource.

At last, I have tested my PR,  and it did work.

Reply:
Yes, I just double check the issue, it's really bug in this scenario, we can have more discussions.

Reply:
What to do next ?  @fisherxu

Reply:
If the edgenode is crashed or reinstalled(data in db lost), the pod will also can't recovered, right? I think we need a design or full solution for it?